### PR TITLE
Update all dependencies

### PR DESCRIPTION
**Pull request by xosmig**
Previously, the project wouldn't build with the latest release of
Go at the time (1.19). The reason for it is that the version of
the quic-go package was not compatible with go 1.19.
Unfortunately, older versions of other packages were not
compatible with the latest version of quic-go, so I just updated
all of them.

### CODE REVIEW

**Comment by xosmig** [comment]
The build in the CI passes without the updated dependencies because the CI is misconfigured (see #185).

File: go.mod [hunk 1]
```diff
@@ -3,50 +3,53 @@ module github.com/filecoin-project/mir
 go 1.18
 
 require (
-	github.com/AlecAivazis/survey/v2 v2.3.2
+	github.com/AlecAivazis/survey/v2 v2.3.5
 	github.com/alecthomas/template v0.0.0-20190718012654-fb15b899a751 // indirect
 	github.com/alecthomas/units v0.0.0-20211218093645-b94a6e3cc137 // indirect
-	github.com/libp2p/go-libp2p v0.19.3
-	github.com/libp2p/go-libp2p-core v0.15.1
+	github.com/libp2p/go-libp2p v0.21.0
```
Comment (dnkolegov) [inline go.mod:9]: Please don't change libp2p dependencies. 
They must be the same as in Eudico - https://github.com/filecoin-project/eudico/blob/eudico/go.mod#L115

File: go.mod [hunk 2]
```diff
@@ -3,50 +3,53 @@ module github.com/filecoin-project/mir
...
+	google.golang.org/protobuf v1.28.1
 	gopkg.in/alecthomas/kingpin.v2 v2.2.6
 )
 
 require github.com/golang/mock v1.6.0
 
 require (
-	github.com/ipfs/go-cid v0.1.0
-	github.com/multiformats/go-multiaddr v0.5.0
+	github.com/ipfs/go-cid v0.2.0
```
Comment (dnkolegov) [inline go.mod:29]: https://github.com/filecoin-project/eudico/blob/eudico/go.mod#L79
Please use 0.1.0

File: go.mod [hunk 3]
```diff
@@ -3,50 +3,53 @@ module github.com/filecoin-project/mir
...
+	github.com/multiformats/go-multiaddr v0.6.0
```
Comment (dnkolegov) [inline go.mod:30]: Please use 0.5.0 - https://github.com/filecoin-project/eudico/blob/eudico/go.mod#L135

**Review by dnkolegov — COMMENTED** [review]
Please don't update libp2p, libp2p-core, go-cid, and similar dependencies that are used in Eudico.

Even if the tests pass here Eudico will not be working with Mir.

See: 
https://github.com/filecoin-project/mir/commit/22b4ef732909d98828d67717564161e51553092c

https://github.com/filecoin-project/mir/commit/3006260b0cef7ba6533fe84771c3b0968f1fd643

I would like to remind (and that is not obvious) that at present Mir must use Go modules and features which are compatible with Eudico.

**Comment by xosmig** [comment]
@dnkolegov the old versions of the libp2p libraries do not work with the latest version of QUIC.
So, I guess, no go 1.19 for us for now, until Eudico updates.

**Comment by dnkolegov** [comment]
> So, I guess, no go 1.19 for us for now, until Eudico updates.

We can discuss this during the team week if necessary. I mean do we need a special Mir version for Eudico.

**Comment by xosmig** [comment]
I, personally, don't really need go 1.19, I just updated it locally due to `gonagci-lint` using gofmt from 1.19.

**Pull request closed without merging**